pkg/operator/operator.go

```diff
@@ -84,16 +91,46 @@ func (c serviceCAOperator) Key() (metav1.Object, error) {
 }
 
 func (c serviceCAOperator) Sync(obj metav1.Object) error {
-	operatorConfig := obj.(*scsv1.ServiceCA)
+	update := false
+	operatorConfig := obj.(*operatorv1.ServiceCA)
 
 	switch operatorConfig.Spec.ManagementState {
-	case scsv1.Unmanaged, scsv1.Removed, "Paused":
+	case operatorv1.Unmanaged, operatorv1.Removed, "Paused":
 		// Totally disable the sync loop in these states until we bump deps and replace sscs.
 		return nil
 	}
+
+	operatorConfigCopy := operatorConfig.DeepCopy()
 	// This is to push out deployments but does not handle deployment generation like it used to. It may need tweaking.
-	err := sync_v4_00_to_latest(c, operatorConfig)
-	return err
+	syncErr := sync_v4_00_to_latest(c, operatorConfigCopy)
+	if syncErr != nil {
+		c.setFailingStatus(operatorConfigCopy, "OperatorSyncLoopError", syncErr.Error())
+	} else {
+		// Set current version and available status
+		version := os.Getenv("RELEASE_VERSION")
```

Comment:
can you confirm that you know that all the pods in openshift-service-ca are all running the latest version at this point and no old pods remain?  that is basically what we are trying to ensure when you set the version.

Reply:
I've updated the status reporting, will test to be sure it has the correct behavior